src/services/auth/register.service.ts

```diff
@@ -0,0 +1,48 @@
+import axios, { AxiosError } from "axios";
+
+type RegisterRequest = {
+  username: string;
+  password: string;
+  confirmPassword: string;
+};
+
+type RegisterResponse = {
+  success: boolean;
+  msg: string;
+  token?: string;
+};
+
+export const registerService = async (
+  req: RegisterRequest
+): Promise<RegisterResponse> => {
+  try {
+    if (req.password !== req.confirmPassword) {
```

Comment:
You don't need to add validations here. All the validations should be handled by the form.